Sources/SwiftParser/Availability.swift

```diff
@@ -253,6 +253,28 @@ extension Parser {
     return self.consumeAnyToken()
   }
 
+  /// Consume the unexpected version token(e.g. integerLiteral, floatingLiteral) until the period no longer appears.
+  private mutating func parseUnexpectedVersionTokens(unexpectedTokens: [RawSyntax] = []) -> RawUnexpectedNodesSyntax? {
+    var copiedUnexpectedTokens = unexpectedTokens
+    var keepGoing: RawTokenSyntax? = nil
+    var loopProgress = LoopProgressCondition()
+    repeat {
+      if let keepGoing {
+        copiedUnexpectedTokens.append(RawSyntax(keepGoing))
+      }
+      if let (_, handle) = self.at(anyIn: VersionTupleWhiteList.self) {
```

Comment:
If you are only checking for integer and float without exhaustively switching over them you can just use `consume(if:)` and don’t need to define a `TokenSpecSubset`

```suggestion
      if versionToken = self.consume(if: .integerLiteral, .floatingLiteral)
```